server/lib/nl/fulfillment/base.py

```diff
@@ -36,10 +36,14 @@
 from server.lib.nl.utterance import QueryType
 from server.lib.nl.utterance import Utterance
 
-# TODO: Factor classification processing functions into a common place.
-
 _EVENT_PREFIX = 'event/'
 
+# Limit the number of charts.  With 3 per row max, allow up to 5 rows.
+_MAX_NUM_CHARTS = 15
```

Comment:
are 3 charts per row fixed? Could this change depending page layout?

Reply:
Oh this is just the rationale for a rough limit.